spring-cloud-kubernetes-fabric8-config/src/main/java/org/springframework/cloud/kubernetes/fabric8/config/Fabric8ConfigUtils.java

```diff
@@ -91,31 +100,143 @@ static String getApplicationNamespace(KubernetesClient client, String namespace,
 
 	}
 
-	/*
-	 * namespace that reaches this point is absolutely present, otherwise this would have
-	 * resulted in a NamespaceResolutionFailedException
+	/**
+	 * <pre>
+	 *     1. read all secrets in the provided namespace
+	 *     2. from the above, filter the ones that we care about
+	 *     3. see if any of the secrets has a single yaml/properties file
+	 *     4. gather all the names of the secrets + decoded data they hold
+	 * </pre>
 	 */
-	static Map<String, String> getConfigMapData(KubernetesClient client, String namespace, String name) {
-		ConfigMap configMap = client.configMaps().inNamespace(namespace).withName(name).get();
+	static Map.Entry<Set<String>, Map<String, Object>> secretsDataByName(KubernetesClient client, String namespace,
+			Set<String> sourceNames, Environment environment,
+			BiFunction<Map<String, String>, Environment, Map<String, Object>> entriesProcessor) {
+		LOG.debug("Reading all secrets in namespace '" + namespace + "'");
+		SecretList secretList = client.secrets().inNamespace(namespace).list();
+		if (secretList == null || secretList.getItems() == null || secretList.getItems().isEmpty()) {
+			LOG.debug("No secrets in namespace '" + namespace + "'");
+			return EMPTY;
+		}
+
+		Set<String> secretNames = new HashSet<>();
+		Map<String, Object> result = new HashMap<>();
+
+		secretList.getItems().stream().filter(secret -> sourceNames.contains(secret.getMetadata().getName()))
+				.collect(Collectors.toList()).forEach(foundSecret -> {
+					String foundSecretName = foundSecret.getMetadata().getName();
+					LOG.debug("Loaded secret with name : '" + foundSecretName + " in namespace: '" + namespace + "'");
+					secretNames.add(foundSecretName);
+
+					Map<String, String> rawData = foundSecret.getData();
+					Map<String, String> decoded = decodeData(rawData == null ? Map.of() : rawData);
+					result.putAll(entriesProcessor.apply(decoded, environment));
+				});
 
-		if (configMap == null) {
-			LOG.warn("config-map with name : '" + name + "' not present in namespace : '" + namespace + "'");
-			return Collections.emptyMap();
+		return Map.entry(secretNames, result);
+
+	}
+
+	/**
+	 * <pre>
+	 *     1. read all secrets in the provided namespace
+	 *     2. from the above, filter the ones that we care about (filter by labels)
+	 *     3. with secret names from (2), find out if there are any profile based secrets (if profiles is not empty)
+	 *     4. concat (2) and (3) and these are the secrets we are interested in
+	 *     5. see if any of the secrets from (4) has a single yaml/properties file
+	 *     6. gather all the names of the secrets (from 4) + data they hold
+	 * </pre>
+	 */
+	static Map.Entry<Set<String>, Map<String, Object>> secretsDataByLabels(KubernetesClient client, String namespace,
```

Comment:
secrets now support single yaml/properties file via `entriesProcessor` - we had such a request in one of the defects, and I saw the opportunity to add it in here, in this PR.